### ISSUE TITLE
CA2237 should only fire if a type have implemented the ISerializable interface directly

### ISSUE THREAD

**Issue by buyaa-n** [dotnet/roslyn-analyzers#6916]
### Analyzer

**Diagnostic ID**: [CA2237](https://learn.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca2237): `Mark ISerializable types with SerializableAttribute`

### Analyzer source

**SDK**: [Built-in CA analyzers in .NET 5 SDK or later](https://learn.microsoft.com/dotnet/fundamentals/productivity/code-analysis)

### Describe the bug

CA2237 should probably be changed to say that if you have implemented the ISerializable interface directly, you should also annotate yourself as [Serializable]. That is, the rule should not fire if the interface is introduced not at your layer, but rather at a layer somewhere else in your type hierarchy. That way you could still subclass ISerializable types and not have to worry about implementing serialization logic yourself.

### Steps To Reproduce

```cs
public class B : A { } // CA2237 should not warn here

[Serializable]
public class A : ISerializable
{
    public void GetObjectData(SerializationInfo info, StreamingContext context)
    {
        throw new NotImplementedException();
    }
}
```

### Expected behavior

CA2237 Should not warn for class B in above example

### Actual behavior

It warns

## Additional context

Originally posting in: https://github.com/dotnet/runtime/issues/90783#issuecomment-1698108338 by @GrabYourPitchforks
